core/backend/src/IModelDb.ts

```diff
@@ -2321,7 +2330,10 @@ export class SnapshotDb extends IModelDb {
   public static openForApplyChangesets(path: LocalFileName, props?: SnapshotOpenOptions): SnapshotDb {
     const file = { path, key: props?.key };
     const nativeDb = this.openDgnDb(file, OpenMode.ReadWrite, undefined, props);
-    return new SnapshotDb(nativeDb, file.key!);
+    if (undefined === file.key) {
```

Comment:
we should change the signature to make key required. @pmconne how do you do that?

Reply:
Shouldn't the key for a snapshot be `path` if `props?.key` is undefined?

Reply:
`openDgnDb` already does that, which is why the "!" was there. Just add an assert.

I now see that the props argument is not used in the one call to this internal function. It should be removed (but I'll do that in another PR).